tutorials/basic-topics/sorting/tim-sort.md

```diff
@@ -11,4 +11,162 @@ keywords:
   - tim sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@Bobliuuu"/>
+
+## Overview
+
+Timsort is a fast stable sorting algorithm based upon both [insertion sort](insertion-sort.md) and [merge sort](merge-sort.md). The algorithm was first created by Tim Peters in 2002, and is now being used in Python’s `sort()` and Java’s `Arrays.sort()`. 
+The reason this algorithm is so fast is because it leverages the benefits of both merge sort and insertion sort. Let’s see how it works!
+
+## Algorithm
+
+Timsort works by first splitting an array into **runs**. A **run** is a subarray of data spliced from the original array. 
+These runs are generated using merge sort (each run has a standard size of 32-64), and insertion sort is used to sort each run. Finally, merge sort combines these sorted arrays together recursively. 
+
+Basically, to run timsort: 
+
+- We split the array into runs.
+- For each run, run insertion sort to sort that section.
+- Merge runs together one by one using merge sort, by comparing values in each sorted list and combining them. 
+
+This algorithm works because each run is sorted using insertion sort, and merge sort makes sure that each subarray is merged to the original array in the correct position. 
+
+## Example: 
+
+> An array of integers in the range [1, n] is given, where each integer appears once or twice. We have to find all the integers that appear twice in the array.
+
+*Naive Approach*: Using a Hashmap or a frequency array, we can store the number of times each element comes up. We then return an array by looping through the frequency array and adding all values that appear twice. However, this requires $$O(n)$$ spaces complexity, and the problem requires us to have $O(1)$ space complexity. 
+
+For this sort of problem, we can use timsort to lower our space complexity! 
+
+### Pseudocode
+
+- Sort the array using timsort
+- Loop through the array
+- If two values are the same, then that value must be repeated. Return that value. 
+
+### Dry Run
+
+Let’s do a dry run of timsort with the array `[5, 4, 3, 1, 2, 6, 7, 4]`, and a run size of 2. . 
+
+- Each run is sorted using insertion sort. The array becomes `[4, 5, 1, 3, 2, 6, 4, 7]`. 
+- The merges happen using recursion. We first attempt to split the array into two parts, down the middle. 
+- First, the left part is merged, meaning the first two runs are merged. Then the array becomes `[1, 3, 4, 5, 2, 6, 4, 7]`. 
+- Then, the right part is merged, meaning the next two runs are merged. The the array becomes `[1, 3, 4, 5, 2, 4, 6, 7]`. 
+- Finally, the entire array is merged. The array is finally sorted: `[1, 2, 3, 4, 4, 5, 6, 7]`.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Bobliuuu"/>
+
+```cpp
+class Solution {
```

Comment:
1. don't use inline comment. put the comment above the line.
2. comments should start with lowercase.